apps/src/code-studio/components/progress/TeacherPanel.jsx

```diff
@@ -76,10 +82,69 @@ class TeacherPanel extends React.Component {
     }),
     scriptHasLockableStages: PropTypes.bool.isRequired,
     unlockedStageNames: PropTypes.arrayOf(PropTypes.string).isRequired,
-    students: PropTypes.arrayOf(studentShape)
+    students: PropTypes.arrayOf(studentType)
   };
 
-  logToFirehose = (eventName, overrideData = {}) => {
+  constructor(props) {
+    super(props);
+    this.logToFirehose = this.logToFirehose.bind(this);
+    this.onSelectUser = this.onSelectUser.bind(this);
+    this.scriptLevels =
+      props.sectionData && props.sectionData.section_script_levels;
+    this.levelProgressByStudent = this.getLevelProgress(this.scriptLevels);
+    this.currentStudent =
+      props.sectionData && this.getCurrentStudent(props.sectionData);
+    this.currentStudentLevel = this.getCurrentStudentLevel(
+      this.currentStudent,
+      this.scriptLevels,
+      props.sectionData
+    );
+  }
+
+  /**
+   * This is the only place in our code where we get levels from
+   * ScriptLevel.summarize_for_teacher_panel or
+   * ScriptLevel.summarize_as_bonus_for_teacher_panel,
```

Comment:
Is this comment still relevant? (asking since summarize_for_teacher_panel etc isn't in this function)

Reply:
haha, this comment is brand new! those summarize methods are in the rails code, which i thought would be clear from the names and mentioning the rails code at the end of this comment, but i can flesh this out with a little more explanation.

Reply:
Ahhh ok. I think I didn't read the whole comment.